shell/src/main/java/tachyon/shell/TfsShell.java

```diff
@@ -174,6 +178,61 @@ public void load(TachyonURI filePath) throws IOException {
   }
 
   /**
+   * Persist a file or directory currently stored only in Tachyon to the UnderFileSystem
+   *
+   * @param filePath the TachyonURI path to persist to the UnderFileSystem
+   * @throws IOException
+   */
+  public void persist(TachyonURI filePath) throws IOException {
+    try {
+      TachyonFile fd = mTfs.open(filePath);
+      FileInfo fInfo = mTfs.getInfo(fd);
+      if (fInfo.isFolder) {
+        List<FileInfo> files = mTfs.listStatus(fd);
```

Comment:
Shouldn't the support of persisting folder be pushed down in `TachyonFileSystem`? I can imagine such functionality can be useful in other places besides shell.